### DISCUSSION TITLE
Logitech G923 PS/PC not recognised by oversteer

### DISCUSSION THREAD

**Discussion by Postboote1** [berarma/new-lg4ff#99] [Q&A] [answered]
Hello 

I wanted to fully switch from Windows to Linux. But therefor the only thing left is getting my wheel to work.
I have installed the lg4ff new for Kernel 6.9, have updated my Kernel to 6.9, installed the linux headers and oversteer.  
I also set the udev rule.
But nothing seems to work. The wheel is still not recognized by oversteer.
However with jstest I could see that the buttons apparently work, but not the axis.
Also followed the arch wiki with following results:
dsmg:

```
[    1.544785] usb 1-7: New USB device found, idVendor=046d, idProduct=c267, bcdDevice=38.00
[    1.544787] usb 1-7: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[    1.544788] usb 1-7: Product: G923 Racing Wheel for PlayStation 4 and PC
[    1.544788] usb 1-7: Manufacturer: Logitech
[    1.544789] usb 1-7: SerialNumber: CEMAWYDILYVYWUPAWOJYTITIDY
[    1.547504] hid-generic 0003:046D:C267.0002: ignoring exceeding usage max
[    1.547594] input: Logitech G923 Racing Wheel for PlayStation 4 and PC as /devices/pci0000:00/0000:00:14.0/usb1/1-7/1-7:1.0/0003:046D:C267.0002/input/input1
[    1.547624] hid-generic 0003:046D:C267.0002: input,hiddev97,hidraw1: USB HID v1.10 Gamepad [Logitech G923 Racing Wheel for PlayStation 4 and PC] on usb-0000:00:14.0-7/input0
[    1.548182] hid-generic 0003:046D:C267.0003: hiddev98,hidraw2: USB HID v1.11 Device [Logitech G923 Racing Wheel for PlayStation 4 and PC] on usb-0000:00:14.0-7/input1
[    1.548688] hid-generic 0003:046D:C267.0004: hiddev99,hidraw3: USB HID v1.11 Device [Logitech G923 Racing Wheel for PlayStation 4 and PC] on usb-0000:00:14.0-7/input2
[    1.607405] usb 1-5.1: new low-speed USB device number 5 using xhci_hcd
```    

than lsusb:

```
Bus 001 Device 011: ID 1e7d:3a37
Bus 001 Device 010: ID 8087:0033
Bus 001 Device 005: ID 046d:c215
Bus 001 Device 004: ID 046d:c267
Bus 001 Device 008: ID 046d:c335
Bus 001 Device 001: ID 1d6b:0002
Bus 001 Device 003: ID 05e3:0608
Bus 001 Device 009: ID 05e3:0608
Bus 002 Device 001: ID 1d6b:0003
Bus 001 Device 006: ID 046d:c07e
Bus 001 Device 002: ID 1462:7e06
``` 

than input cat /proc/bus/input/devices

```
I: Bus=0003 Vendor=046d Product=c267 Version=0110
N: Name="Logitech G923 Racing Wheel for PlayStation 4 and PC"
P: Phys=usb-0000:00:14.0-7/input0
S: Sysfs=/devices/pci0000:00/0000:00:14.0/usb1/1-7/1-7:1.0/0003:046D:C267.0002/input/input1
U: Uniq=CEMAWYDILYVYWUPAWOJYTITIDY
H: Handlers=event1 js0 
B: PROP=0
B: EV=1b
B: KEY=3fff000000000000 0 0 0 0
B: ABS=3003f
B: MSC=10
```
 
than jstest:

```
Driver version is 2.1.0.
Joystick (Logitech G923 Racing Wheel for PlayStation 4 and PC) has 8 axes (X, Y, Z, Rx, Ry, Rz, Hat0X, Hat0Y)
and 14 buttons (BtnA, BtnB, BtnC, BtnX, BtnY, BtnZ, BtnTL, BtnTR, BtnTL2, BtnTR2, BtnSelect, BtnStart, BtnMode, BtnThumbL).
Testing ... (interrupt to exit)
Axes:  0:     0  1:     0  2:     0  3:-32767  4:-32767  5:     0  6:     0  7:     0 Buttons:  0:off  1:off  2:off  3:off  4:off  5:off  6:off  7:off  8:off  9:off 10:off 11:off 12:off 13:off 
```

I am currently using NixOS 23.11 with Kernel Version 6.9.4.

Please help thank you

**Comment by berarma** [comment]
It seems new-lg4ff isn't correctly installed. Please, try following the installation steps again. Once installed you should reboot your computer.

When new-lg4ff is loaded, the USB product id for the device should change from c267 to c266 and a hid-logitech message should appear in dmesg.

**Reply by Postboote1** [reply]
The problem is that I use Nix OS. To install the driver I simply put the package name oft the driver in my configuration.nix and this should work.
Could it be that the udev rule needed isn't doing it's job?

**Reply by berarma** [reply]
> The problem is that I use Nix OS. To install the driver I simply put the package name oft the driver in my configuration.nix and this should work.

I can't say about this. Any issue related to the installation on Nix OS would be out of my reach.

> Could it be that the udev rule needed isn't doing it's job?

No, that's only needed for permissions to change settings.

**Reply by Postboote1** [reply, marked as answer]
So the problem was my Logitech Extrem 3d Pro Joystick.

After I unpluged it the wheel was recognised.

**Reply by berarma** [reply]
Good to know. Thanks for posting.
